add debug doc for rudr (#458)
oam-dev/rudr: docs/developer/debug.md

# How to Debug?

> Now Rudr is still in heavily develop, if you have troubles playing with it.

Reviewer: ```suggestion
```

> You could try to debug with following steps.

Reviewer: ```suggestion
This document outlines troubleshooting guides to help you debug things when deployments via Rudr are not successful.```

## Check Rudr pod is running

> Make sure rudr is running successfully.

Reviewer: We should be consisting with the casing for Rudr.

 
```
$ kubectl get pods -l app.kubernetes.io/name=rudr
NAME                    READY   STATUS    RESTARTS   AGE
rudr-6b9b9c57cd-zxgfr   1/1     Running   0          30s
```

## Check Rudr logs

When you deploy an Application Configuration, you could find information about it from logs.

```
$ kubectl logs -f rudr-6b9b9c57cd-zxgfr 
```

Currently, all workload implemented in Rudr will create pods.

> So if your Application Configuration didn't create any pod, there must be some error information in the logs.

Reviewer: ```suggestion
So if your ApplicationConfiguration didn't create any pod, there must be some error information in the logs.
```

> Check the logs and report to us.

Reviewer: ```suggestion
Check the logs and file an issue with the log information. 
```

> ## Check Component exist

Reviewer: ```suggestion
## Check if the Component exists
```

```shell script
$ kubectl get comp
NAME                         AGE
alpine-singleton-task-v1     3s
alpine-singleton-worker-v1   3s
alpine-task-v1               3s
alpine-worker-v1             3s
hpa-example-replicated-v1    3s
nginx-replicated-v1          3s
nginx-singleton-v1           3s
```

Don't forget to deploy component before using them.

## Check parameters

> Check the parameters and properties of Application Configuration are consistent with trait and component.   

Reviewer: ```suggestion
Check the parameters and properties of ApplicationConfiguration are consistent with trait and component.   
```
